### ISSUE TITLE
order-properties: consider an option for sort-package-json's order

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/eslint-plugin-package-json#31]
Right now, the `package-json/order-properties` rule uses its own "standard order": https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/b6553eca1e314d9def74e37af94913785804d26e/lib/rules/order-properties.js#L16

I haven't seen this order in recent history. But, I _have_ seen https://github.com/keithamus/sort-package-json be used in recent ecosystem projects such as https://github.com/matzkoh/prettier-plugin-packagejson. Perhaps `order-properties` could take in an option?

cc @zetlen - is there a good reference for the current order? I'm wondering what to name if it there is an option added...

cc @kellyselden from https://github.com/kellyselden/eslint-plugin-json-files/blob/3a24e6f36b7828210a56070af4487cb5831f7f24/docs/rules/sort-package-json.md - I do find it interesting that both of these ESLint plugins have package.json sorting implemented...

**Comment by Zamiell** [comment]
I think that most of the ecosystem uses `sort-package-json`.
I would vote that we match the ordering of that by default.

**Comment by zetlen** [comment]
I think I took the ordering from the order in which [the documentation page describes them.](https://docs.npmjs.com/cli/v10/configuring-npm/package-json) It doesn't say anywhere that this is a _required_ order. If there's another, more widespread convention, use that.

**Comment by JoshuaKGoldberg** [comment]
Awesome! Let's go with, say, `order: "legacy" | "sort-package-json"``? With `"legacy"` as the default now, noted to switch to `"sort-package-json"` in a potential future version?

Marking as accepting PRs. If nobody's sent a PR by the time I've overhauled this repo (#33 through #37) I'll go ahead and send one myself.

Note that adding `sort-package-json` as a dependency might clobber `package-lock.json` until #33 is in. I've found that using `npx npm@6` to run an older version of npm is a workaround.
